packages/zowe-explorer/src/dataset/ZoweDatasetNode.ts

```diff
@@ -455,10 +455,10 @@ export class ZoweDatasetNode extends ZoweTreeNode implements IZoweDatasetTreeNod
      *
      * @returns {imperative.IProfileLoaded}
      */
-    public getProfile(): imperative.IProfileLoaded {
-        const prof = this.profile ?? this.getParent()?.getProfile();
-
-        return Profiles.getInstance().loadNamedProfile(prof.name); // this returns the profile with newer token
+    public getProfileInstance(): imperative.IProfileLoaded {
+        ZoweLogger.trace("ZoweDatasetNode.getProfileInstance called.");
+        const prof = this.getProfile();
+        return prof.name ? Profiles.getInstance().loadNamedProfile(prof.name) : undefined; // this returns the profile with newer token
     }
```

Comment:
I noticed that the function implementation was renamed to `getProfileInstance`. Is this change intentional? We currently use the `getProfile` function in several other places in the context of data set nodes - I think it would be best to implement the fix within the `getProfile` implementation itself so that the fix is applied everywhere. Otherwise, we'll need to update some references in the `dataset/actions.ts` file so that it refers to the proper profile.

Reply:
The original fix was causing a bunch of test issues and since this fix was relatively for opening of datasets and getting the children, I decided to change up the name and make it specific to just this class but still call the getProfile name. But, the original getProfile is still there in packages/zowe-explorer-api/src/tree/ZoweTreeNode.ts, so other classes shouldn't be affected. At first I was unable to import the "Profiles" to the original getProfile method, otherwise I think we would see less test failures.

Reply:
After discussing with Billie, we will stick to the getProfile Name and I will work on the failing test cases.